src/shared/awsagent/agent.go

```diff
@@ -18,20 +26,31 @@ type Agent struct {
 
 func NewAWSAgent(
 	ctx context.Context,
-	oidcUrl string, // TODO: use eks.DescribeCluster to get the OIDC URL.
+	oidcUrlFromEnv string,
 ) *Agent {
 	logrus.Info("AWS Intents agent - enabled")
 
-	if oidcUrl == "" {
-		logrus.Fatal("EKS cluster OIDC url not provided")
-	}
-
 	awsConfig, err := config.LoadDefaultConfig(context.Background())
 
 	if err != nil {
 		panic(err)
 	}
 
+	oidcUrl := ""
+
+	if oidcUrlFromEnv == "" {
+		retrieved, err := tryRetrieveOIDCUrl(ctx, awsConfig)
+
+		if err != nil {
+			logrus.WithError(err).Fatal("OIDC URL not provided, and not retrieved")
```

Comment:
@orishavit - let's stick to returning errors and only exiting in main on failure. should not use fatal here